docs/website/docs/general-usage/credentials/config_specs.md

```diff
@@ -0,0 +1,327 @@
+desktop applications in Google Cloud Platform (GCP).
+It can parse native values either as GoogleOAuth2Credentials or as
+serialized OAuth client secrets JSON.
+This class provides methods for authentication and obtaining access tokens.
+
+Usage:
+```python
+oauth_credentials = GcpOAuthCredentials()
+
+# Accepts a native value, which can be either an instance of GoogleOAuth2Credentials
+# or serialized OAuth client secrets JSON.
+# Parses the native value and updates the credentials.
+native_value_oauth = ...
+oauth_credentials.parse_native_representation(native_value_oauth)
+
+# Authenticate and get access token
+oauth_credentials.auth(scopes=["scope1", "scope2"])
+
+# Retrieve native credentials for Google clients
+# For example, build the service object for Google Analytics PI.
+client = BetaAnalyticsDataClient(credentials=oauth_credentials.to_native_credentials())
+
+# Get a string representation of the credentials
+# Returns a string representation of the credentials in the format client_id@project_id.
+credentials_str = str(oauth_credentials)
+```
+
+[The example](https://github.com/dlt-hub/verified-sources/blob/master/sources/google_analytics/setup_script_gcp_oauth.py): how you can get the refresh token using `dlt.secrets.value`.
+
+### AwsCredentials
+
+The AwsCredentials class is responsible for handling AWS credentials,
+including access keys, session tokens, profile names, region names, and endpoint URLs.
+It inherits the ability to manage default credentials and extends it with methods
+for handling partial credentials and converting credentials to a botocore session.
+
+Usage:
+```python
+credentials = AwsCredentials()
+
+# Set the necessary attributes
+credentials.aws_access_key_id = "ACCESS_KEY_ID"
+credentials.aws_secret_access_key = "SECRET_ACCESS_KEY"
+credentials.region_name = "us-east-1"
+
+# Convert credentials to s3fs format
+s3fs_credentials = credentials.to_s3fs_credentials()
+print(s3fs_credentials["key"])
+
+# Imports an external boto3 session and sets the credentials properties accordingly.
+import botocore.session
+session = botocore.session.get_session()
+credentials.parse_native_representation(session)
+print(credentials.aws_access_key_id)
+
+# Get AWS credentials from botocore session
+aws_credentials = credentials.to_native_credentials()
+print(aws_credentials.access_key)
+```
+
+### AzureCredentials
+
+The AzureCredentials class is responsible for handling Azure Blob Storage credentials,
+including account name, account key, Shared Access Signature (SAS) token, and SAS token permissions.
+It inherits the ability to manage default credentials and extends it with methods for
+handling partial credentials and converting credentials to a format suitable
+for interacting with Azure Blob Storage using the adlfs library.
+
+Usage:
+```python
+credentials = AzureCredentials()
+
+# Set the necessary attributes
+credentials.azure_storage_account_name = "ACCOUNT_NAME"
```

Comment:
same thing!
we detect default credentials as well by calling `DefaultAzureCredential`. `to_native_credentials()` is not yet implemented